apis/v1beta1/gateway_types.go

```diff
@@ -579,76 +594,54 @@ const (
+	// the Gateway.
+	GatewayReasonPending GatewayConditionReason = "Pending"
+
+	// This reason is used with the "Accepted" condition when the Gateway could not be configured
+	// because the requested address is not supported. This reason could be used in a number of
+	// instances, including:
+	//
+	// * The address is already in use.
+	// * The type of address is not supported by the implementation.
+	GatewaReasonUnsupportedAddress GatewayConditionReason = "UnsupportedAddress"
+)
+
+const (
+	// Deprecated: use "Accepted" instead.
+	GatewayConditionScheduled GatewayConditionType = "Scheduled"
+
 	// This reason is used with the "Scheduled" condition when the condition is
 	// True.
 	//
 	// Deprecated: use the "Accepted" condition with reason "Accepted" instead.
 	GatewayReasonScheduled GatewayConditionReason = "Scheduled"
 
-	// This reason is used with the "Accepted", "Programmed" and "Ready"
-	// conditions when the status is "Unknown" and no controller has reconciled
-	// the Gateway.
-	GatewayReasonPending GatewayConditionReason = "Pending"
-
 	// Deprecated: Use "Pending" instead.
 	GatewayReasonNotReconciled GatewayConditionReason = "NotReconciled"
-
-	// This reason is used with the "Accepted" condition when the
-	// Gateway is not scheduled because insufficient infrastructure
-	// resources are available.
-	GatewayReasonNoResources GatewayConditionReason = "NoResources"
 )
 
 const (
-	// Ready is an optional Condition that has Extended support. When it's set,
-	// the condition indicates whether the Gateway has been completely configured
-	// and traffic is ready to flow through the data plane immediately.
-	//
-	// If both the "ListenersNotValid" and "ListenersNotReady"
-	// reasons are true, the Gateway controller should prefer the
-	// "ListenersNotValid" reason.
-	//
-	// Possible reasons for this condition to be true are:
+	// "Ready" is a reserved condition type for future use. It should not be used by implementations.
+	// Note: its not really "deprecated", but rather "reserved"; however, deprecated triggers Go linters
+	// to alert about usage.
 	//
-	// * "Ready"
-	//
-	// Possible reasons for this condition to be False are:
-	//
-	// * "ListenersNotValid"
-	// * "ListenersNotReady"
-	// * "AddressNotAssigned"
-	//
-	// Controllers may raise this condition with other reasons,
-	// but should prefer to use the reasons listed above to improve
-	// interoperability.
+	// Deprecated: Ready is reserved for future use
 	GatewayConditionReady GatewayConditionType = "Ready"
 
-	// This reason is used with the "Ready" condition when the condition is
-	// true.
+	// Deprecated: Ready is reserved for future use
 	GatewayReasonReady GatewayConditionReason = "Ready"
 
-	// This reason is used with the "Ready" condition when one or
-	// more Listeners have an invalid or unsupported configuration
-	// and cannot be configured on the Gateway.
+	// Deprecated: Ready is reserved for future use
```

Comment:
This feels like it should move to "Programmed"?

Reply:
I mentioned this in https://github.com/kubernetes-sigs/gateway-api/issues/1832 but left it out since it seemed controversial. 

I will add it if folks want

Reply:
I liked the way you specified it in https://github.com/kubernetes-sigs/gateway-api/issues/1832#issuecomment-1487167378. I think it could be fine as long as we have the "can be present when Accepted is True" caveat.

Reply:
Added it back